### ISSUE TITLE
Clang-tidy: modernize-use-nullptr produces a false-positive on std:: strong_ordering (spaceship operator result)

### ISSUE THREAD

**Issue by fastasturtle** [llvm/llvm-project#63478]
```cpp
#include <compare>

int main() {
    auto res = 0 <=> 1;
    if (res < 0) {} // warning: use nullptr [modernize-use-nullptr]
}
```

https://gcc.godbolt.org/z/89W17v8j7

**Comment by PiotrZSL** [comment]
I'm not sure if this is false positive, simply if you put there any other number than 0, it wont compile, and in AST this 0 is implicit casted to pointer during construction of `__cmp_cat::__unspec`.

**Comment by fastasturtle** [comment]
> I'm not sure if this is false positive, simply if you put there any other number than 0, it wont compile, and in AST this 0 is implicit casted to pointer during construction of `__cmp_cat::__unspec`.

@PiotrZSL while this is technically possible to use `nullptr` instead of 0, the comparison operator return value result is expected to be compared with 0 (similar to the return value of `strcpy`, etc) - see e.g. the [original proposal](https://open-std.org/jtc1/sc22/wg21/docs/papers/2017/p0515r0.pdf).

The way it's worded in the standard is (if I'm reading it correctly) is here: https://eel.is/c++draft/cmp#categories.pre-3

**Comment by PiotrZSL** [comment]
https://reviews.llvm.org/D158928
